### PR TITLE
Add support for black code formatter

### PR DESCRIPTION
**Pull request by aabadie**
This PR integrate the use of the [black](https://black.readthedocs.io/en/stable/) code formatter in the CI checks (tox env + gh actions).
All changes requested by black are applied in a single extra commit (using `black .` locally).

This PR is based on #26 (and #25 indirectly)

### CODE REVIEW

**Comment by aabadie** [comment]
rebased!

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
     - name: Test with tox
       run: |
         pyenv="py$(echo "${{ matrix.python-version }}" | tr -d '.')"
-        tox -e ${pyenv}-test,${pyenv}-rapidjson,flake8,lint
+        tox -e ${pyenv}-test,${pyenv}-rapidjson,flake8,lint,black
```
Comment (miri64) [inline .github/workflows/tox.yml:34]: Appearently this also needs to be handled separately, as `black` seems to have dropped 3.5 support ...

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = {py35,py36,py37,py38,py39}-{test,rapidjson},lint,flake8,check_package
+envlist = {py35,py36,py37,py38,py39}-{test,rapidjson},lint,flake8,check_package,black
```
Comment (miri64) [inline tox.ini:2]: I'm fine with having static linting done _after_ testing, but changing the code after testing is just mean ;-).

**Review by miri64 — CHANGES_REQUESTED** [review]

File: setup.cfg [hunk 3]
```diff
@@ -9,9 +9,14 @@ markers =
 
 [pylint]
 reports = no
-disable = locally-disabled,star-args,consider-using-f-string
+disable = line-too-long,locally-disabled,star-args,consider-using-f-string
```
Comment (miri64) [inline setup.cfg:12]: How is this related? I really do not like too long lines.

File: setup.cfg [hunk 4]
```diff
@@ -9,9 +9,14 @@ markers =
...
 msg-template = {path}:{line}: [{msg_id}({symbol}), {obj}] {msg}
 
 [flake8]
 exclude = .tox,dist,doc,build,*.egg
 max-complexity = 10
+ignore = E501, W503  # line too long and line break before binary operator
```
Comment (miri64) [inline setup.cfg:18]: Same here.

File: setup.cfg [hunk 5]
```diff
@@ -9,9 +9,14 @@ markers =
...
+
+[tool.black]
+line-length = 79
```
Comment (miri64) [inline setup.cfg:21]: Ok, we set it to 79 here, but why can't the other linters still have this check?

**Review by miri64 — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 2 above)
Comment (aabadie) [reply tox.ini:2]: The current check doesn't change any code, it just verifies if code should be changed and if yes it prints the diff. You can locally run `black .` to apply auto-formatting.

File: .github/workflows/tox.yml (hunk 1 above)
Comment (aabadie) [reply .github/workflows/tox.yml:34]: Yes unfortunately...

File: setup.cfg (hunk 3 above)
Comment (aabadie) [reply setup.cfg]: I took this from the configuration of another of my Python package. But after re-reading the manual, it seems that it could be simplified. Interesting read: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html

File: setup.cfg (hunk 5 above)
Comment (aabadie) [reply setup.cfg]: Because black is not strict with line-length, it tries to follow it but if it can't because it breaks other formatting rules, it will reformat with a larger line-length, see https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length

**Review by miri64 — APPROVED** [review]
